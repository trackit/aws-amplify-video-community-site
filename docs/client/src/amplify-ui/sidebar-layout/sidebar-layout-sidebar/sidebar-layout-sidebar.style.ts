import {css} from "emotion";

export const sidebarLayoutSidebarStyle = css`
  display: none;
  border-right: 0.0625rem solid var(--border-color);
  min-width: 20rem;

  &.in-view {
    display: initial;
  }

  > div {
    position: relative;
    min-height: 100vh;
    height: 100%;

    > div {
      position: sticky;
      display: flex;
      flex-direction: column;
      max-height: 100vh;
      overflow-y: auto; /* for Firefox */
      overflow-y: overlay; /* for Webkit browsers */
      margin-bottom: 6rem;
    }
  }
`;
