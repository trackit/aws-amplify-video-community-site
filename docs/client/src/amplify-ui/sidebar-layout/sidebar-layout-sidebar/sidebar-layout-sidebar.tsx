import {Component, Host, Listen, Prop, h} from "@stencil/core";
import {sidebarLayoutContext} from "../sidebar-layout.context";
import {sidebarLayoutSidebarStyle} from "./sidebar-layout-sidebar.style";
import {ToggleInView} from "../sidebar-layout.types";
import {MQTablet} from "../../styles/media";

@Component({tag: "amplify-sidebar-layout-sidebar", shadow: false})
export class AmplifySidebarLayoutSidebar {
  /*** is the sidebar hidden? This prop is passed via `sidebar-context` */
  @Prop() readonly inView?: boolean;
  /*** at what distance from the viewport top should the bar be affixed? */
  @Prop() readonly top?: number;
  /*** toggles the state provided by `sidebar-layout` */
  @Prop() readonly toggleInView?: ToggleInView;

  ref?: HTMLElement | null;
  setRef = (ref: HTMLElement | null) => (this.ref = ref);

  @Listen("click", {target: "document"})
  clickListener(e: Event) {
    // Slice off the "@media " string at the start for use in JS instead of CSS
    const MQTabletJS = MQTablet.substring(6);
    const isMobile = !window.matchMedia(MQTabletJS).matches;
    // Hide the sidebar only if we're on mobile and the user tapped the content, not the sidebar itself
    if (isMobile && this.inView) {
      const clickedElement = e.target as HTMLElement | undefined;
      if (this.ref && clickedElement && !this.ref.contains(clickedElement)) {
        this.toggleInView && this.toggleInView();
      }
    }
  }

  render() {
    return (
      <Host
        ref={this.setRef}
        class={{
          "in-view": this.inView === undefined || this.inView,
          [sidebarLayoutSidebarStyle]: true,
        }}
      >
        <div>
          <div
            style={{
              top:
                this.top === undefined ? "initial" : `${String(this.top)}rem`,
            }}
          >
            <slot />
          </div>
        </div>
      </Host>
    );
  }
}

sidebarLayoutContext.injectProps(AmplifySidebarLayoutSidebar, [
  "inView",
  "toggleInView",
]);
