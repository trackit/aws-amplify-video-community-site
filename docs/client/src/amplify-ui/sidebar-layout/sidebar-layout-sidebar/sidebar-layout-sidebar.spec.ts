import {AmplifySidebarLayoutSidebar} from "./sidebar-layout-sidebar";
import {newSpecPage} from "@stencil/core/testing";

describe("amplify-sidebar-layout-sidebar", () => {
  describe("Component logic", () => {
    let amplifySidebarLayoutSidebar: AmplifySidebarLayoutSidebar;
    beforeEach(
      () => (amplifySidebarLayoutSidebar = new AmplifySidebarLayoutSidebar()),
    );

    it("should init `toggleInView` as `undefined`", () =>
      expect(amplifySidebarLayoutSidebar.toggleInView).toBeUndefined());

    it("should init `inView` as `undefined`", () =>
      expect(amplifySidebarLayoutSidebar.inView).toBeUndefined());

    it("should init `top` as `undefined`", () =>
      expect(amplifySidebarLayoutSidebar.top).toBeUndefined());
  });

  describe("Render logic", () => {
    it("should render", async () => {
      expect(
        (
          await newSpecPage({
            components: [AmplifySidebarLayoutSidebar],
            html: `<amplify-sidebar-layout-sidebar />`,
          })
        ).root,
      ).toMatchSnapshot();
    });
  });
});
