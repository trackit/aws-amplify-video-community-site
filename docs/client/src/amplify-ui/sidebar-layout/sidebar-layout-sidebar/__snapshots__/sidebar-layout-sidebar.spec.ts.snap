// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-sidebar-layout-sidebar Render logic should render 1`] = `
<amplify-sidebar-layout-sidebar class="css-kdiujo in-view">
  <!---->
  <div>
    <div style="top: initial;"></div>
  </div>
</amplify-sidebar-layout-sidebar>
`;
